crates/serialize_hierarchy_derive/src/lib.rs

```diff
@@ -266,6 +287,26 @@ fn parse_attributes(attrs: &[syn::Attribute]) -> HashSet<TypeAttribute> {
         .flat_map(parse_meta_items)
         .map(|meta| match meta {
             NestedMeta::Meta(Meta::Path(word)) if word.is_ident("as_jpeg") => TypeAttribute::AsJpeg,
+            NestedMeta::Meta(Meta::NameValue(MetaNameValue {
+                path, lit: literal, ..
+            })) if path.is_ident("bound") => {
```

Comment:
Instead of acquiring the trait bounds as string, couldn't you just parse the trait bounds on the struct? Yes, that would mean that you will have more strict trait bounds on the struct if you are not using the `SerializeHierarchy` trait but I think there is a trade-off between more strict trait bounds and a string Freitextfeld.

Reply:
The string is not a string. It is parsed as a list of trait bounds and reports the correct errors. I think this is already very pretty.
The trait bounds on the struct are too strict. The bounds should only be on the respective implementation

Reply:
In the future, we might be able to automatically guess the appropriate bounds and the user does not have to pass this attribute anymore (like `serde` does)

Reply:
Ok, I predicted your answer. Fine for me.